proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -1,176 +1,219 @@
-          * log on warning level with «unsupported_values» (enum’s/struct’s names)
-    
-  **Note**: For `RegisterAppInterface` to filter we should use version of PROTOCOL, but not the MOBILE_API according to system’s flow. **iOS** *should be extended with similar mechanism in order to encapsulate filtering logic.*
+     * If there are no enum values unknown to SDL, the proxy forwards the request to SDL Core without changes.
+
+     * If some of the enum values are unknown to SDL Core the proxy should:
+          * Filter out all RPC’s unsupported enum values including nested RPC’s if those are present, before making a request to make sure it is compliant with the current spec
+          * Log on warning level with «unsupported_values» (enum’s/struct’s names)
 
-  **For example** : 
-  If the UI app requested unsupported `AppHMIType` and version which are not compatible then the proxy must filter unsupported HMI types. 
+  **Note**: For `RegisterAppInterface` to filter we should use version of PROTOCOL, but not the MOBILE_API according to system’s flow. **iOS** should be extended with similar mechanism in order to encapsulate filtering logic.
+
+  **For example** :
+  If the UI app requested unsupported `AppHMIType` and version which are not compatible, then the proxy must filter unsupported HMI types.
 
       - Before filtering :
           "AppHMIType" : ["DEFAULT", "MEDIA", "UKNOWN"]
       - After filtering :
           AppHMIType" : ["DEFAULT", "MEDIA"]
+
   ***Note**: In case no HMI types are left after filtering, Mobile library should throw Exception, otherwise RegisterAppInterface RPC will be sent with remaining HMI types.*
+
   ***Note**: Now system flow is the following: if `RegisterAppInterface` fails with an error then Mobile library should pass the error status to the Mobile Library adding failure reason*
+
   ***Note**: The protocol version control logic partly implemented in Android and not implemented in iOS though it has its flaws and should be updated following the logic above.*
 
-     
-  ## Potential downsides
+
+## Potential downsides
   There might be certain corner cases with filtering out RPCs appeared since version greater than current.
-  *Note: For `RegisterAppInterface` we cannot keep connection if invalid param is present due to the current flow of the older SDL versions. The connection can be dropped (End Session) by SDL core but not by Mobile Device (SDL library)*
-      
-  ## Impact on existing code
-  Android/iOS 
+
+  **Note:** For `RegisterAppInterface` we cannot keep connection if invalid param is present due to the current flow of the older SDL versions. The connection can be dropped (End Session) by SDL Core but not by Mobile Device (SDL library)
+
+## Impact on existing code
+  **Android/iOS**
+
      * Each `RPCStruct` should be extended with the implementation of filtering mechanism
      * Implement logging for unsupported values
```

Comment:
This is now showing up as a code block, perhaps due to the indenting? Please fix formatting.